File: backend/app/schemas/email_detail.py
"""
Pydantic schemas for the Email Detail API.
"""

from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class EmailDetailResponse(BaseModel):
    """Complete stored information for an ingested email."""

    email_id: str
    message_id: Optional[str]
    matter_key: Optional[str]
    sender: str
    to_recipients: Optional[List[dict]]
    cc_recipients: Optional[List[dict]]
    subject: Optional[str]
    body_text: Optional[str]
    received_at: Optional[datetime]
    raw_file_path: str
    content_hash: str
    processing_status: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/emailDetail.ts
export interface EmailDetail {
  email_id: string;
  message_id: string | null;
  matter_key: string | null;
  sender: string;
  to_recipients: Array<Record<string, unknown>> | null;
  cc_recipients: Array<Record<string, unknown>> | null;
  subject: string | null;
  body_text: string | null;
  received_at: string | null;
  raw_file_path: string;
  content_hash: string;
  processing_status: string;
  created_at: string;
  updated_at: string;
}
